planner/core/find_best_task.go

```diff
@@ -2377,6 +2375,22 @@ func (ds *DataSource) convertToPointGet(prop *property.PhysicalProperty, candida
 		if partitionInfo == nil {
 			return invalidTask
 		}
+	} else if pi != nil && ds.SCtx().GetSessionVars().StmtCtx.UseDynamicPruneMode {
```

Comment:
Can UseDynamicPruneMode be false here (if pi != nil)? Would not that be inconsistent with the setting of ds.isPartition above?

Reply:
Updated